test/vesta/LiquidationBonusTest.js

```diff
@@ -0,0 +1,217 @@
+const { assert } = require("hardhat")
+const deploymentHelper = require("../../utils/deploymentHelpers.js")
+const testHelpers = require("../../utils/testHelpers.js")
+const TroveManagerTester = artifacts.require("./TroveManagerTester.sol")
+const StabilityPool = artifacts.require("./StabilityPool.sol")
+const VSTTokenTester = artifacts.require("./VSTTokenTester.sol")
+
+const th = testHelpers.TestHelper
+const dec = th.dec
+const toBN = th.toBN
+
+contract('TroveManager', async accounts => {
+  const BONUS_5_PERCENT = '50000000000000000'
+  const ICR_103_PERCENT = '103000000000000000000'
+  const ICR_107_PERCENT = '107000000000000000000'
+  const ZERO_ADDRESS = th.ZERO_ADDRESS
+
+  const [owner, alice, whale] = accounts;
+
+  const multisig = accounts[999]
+
+  let priceFeed
+  let VSTToken
+  let sortedTroves
+  let troveManager
+  let activePool
+  let stabilityPool
+  let collSurplusPool
+  let defaultPool
+  let borrowerOperations
+  let vestaParameters
+  let hintHelpers
+  let erc20
+
+  let contracts
+
+  const openTrove = async (params) => th.openTrove(contracts, params)
+
+  beforeEach(async () => {
+    contracts = await deploymentHelper.deployLiquityCore()
+    contracts.troveManager = await TroveManagerTester.new()
+    contracts.vstToken = await VSTTokenTester.new(
+      contracts.troveManager.address,
+      contracts.stabilityPoolManager.address,
+      contracts.borrowerOperations.address
+    )
+    const VSTAContracts = await deploymentHelper.deployVSTAContractsHardhat(accounts[0])
+
+    priceFeed = contracts.priceFeedTestnet
+    VSTToken = contracts.vstToken
+    sortedTroves = contracts.sortedTroves
+    troveManager = contracts.troveManager
+    activePool = contracts.activePool
+    defaultPool = contracts.defaultPool
+    collSurplusPool = contracts.collSurplusPool
+    borrowerOperations = contracts.borrowerOperations
+    hintHelpers = contracts.hintHelpers
+    vestaParameters = contracts.vestaParameters;
+
+    vstaStaking = VSTAContracts.vstaStaking
+    vstaToken = VSTAContracts.vstaToken
+    communityIssuance = VSTAContracts.communityIssuance
+
+    await vstaToken.unprotectedMint(multisig, dec(1, 24))
+
+    erc20 = contracts.erc20;
+
+    let index = 0;
+    for (const acc of accounts) {
+      await vstaToken.approve(vstaStaking.address, await web3.eth.getBalance(acc), { from: acc })
+      await erc20.mint(acc, await web3.eth.getBalance(acc))
+      index++;
+
+      if (index >= 20)
+        break;
+    }
+
+    await deploymentHelper.connectCoreContracts(contracts, VSTAContracts)
+    await deploymentHelper.connectVSTAContractsToCore(VSTAContracts, contracts)
+
+    stabilityPool = await StabilityPool.at(await contracts.stabilityPoolManager.getAssetStabilityPool(ZERO_ADDRESS))
+  })
+
+  it('liquidate(): check borrower and pools when debt(2000) > SP(1000) and ICR(103%) <= 100 + BONUS(5%)', async () => {
```

Comment:
I know the unit tests of Liquity is a mess, but let's try to keep the same logic that we are using at vesta, so what are we expecting from this test.

e.g: No returns to borrower.

`liquidate():  when debt(2000) > SP(1000) and ICR(103%) <= 100 + BONUS(5%), sends collateral to liquidator and stability pool only'`

or

`liquidate():  when debt(2000) > SP(1000) and ICR(103%) <= 100 + BONUS(5%), returns nothing for borrower and pay liquidator and stability pool'`


Because "check borrower and pools" doesn't tell me what this test is doing, what are you checking? What the check should look like? How can I tell this test is valid by it's name? If the test breaks in one of the patch, how can I fix it if I doesn't know what it should check.

Of course, this can be all answered if we read the code. But it's a waste of time for other developers and a chance of confusion.